windows/RNSVG/SvgView.cpp

```diff
@@ -5,36 +5,66 @@
 #include "SvgView.g.cpp"
 #endif
 
+#include <windows.ui.xaml.media.dxinterop.h>
+#include <winrt/Windows.UI.Xaml.Media.Imaging.h>
 #include <winrt/Windows.Graphics.Display.h>
 
+#include "D2DDevice.h"
+#include "D2DDeviceContext.h"
 #include "GroupView.h"
 #include "Utils.h"
 
+#include <d3d11_4.h>
+
 using namespace winrt;
-using namespace Microsoft::Graphics::Canvas;
 using namespace Microsoft::ReactNative;
-using namespace Windows::Graphics::Display;
 
 namespace winrt::RNSVG::implementation {
 SvgView::SvgView(IReactContext const &context) : m_reactContext(context) {
-  m_scale = static_cast<float>(DisplayInformation::GetForCurrentView().ResolutionScale()) / 100;
-
-  m_canvasDrawRevoker = m_canvas.Draw(winrt::auto_revoke, {get_weak(), &SvgView::Canvas_Draw});
-  m_canvasCreateResourcesRevoker = m_canvas.CreateResources(winrt::auto_revoke, {get_weak(), &SvgView::Canvas_CreateResources});
-  m_canvasSizeChangedRevoker = m_canvas.SizeChanged(winrt::auto_revoke, {get_weak(), &SvgView::Canvas_SizeChanged});
+  uint32_t creationFlags = D3D11_CREATE_DEVICE_BGRA_SUPPORT;
+
+  D3D_FEATURE_LEVEL featureLevels[] = {
+      D3D_FEATURE_LEVEL_11_1,
+      D3D_FEATURE_LEVEL_11_0,
+      D3D_FEATURE_LEVEL_10_1,
+      D3D_FEATURE_LEVEL_10_0,
+      D3D_FEATURE_LEVEL_9_3,
+      D3D_FEATURE_LEVEL_9_2,
+      D3D_FEATURE_LEVEL_9_1};
+
+  // Create the Direct3D device.
+  com_ptr<ID3D11Device> d3dDevice;
+  D3D_FEATURE_LEVEL supportedFeatureLevel;
+  check_hresult(D3D11CreateDevice(
+      nullptr, // default adapter
+      D3D_DRIVER_TYPE_HARDWARE,
+      0,
+      creationFlags,
+      featureLevels,
+      ARRAYSIZE(featureLevels),
```

Comment:
```suggestion
      sizeof(featureLevels),
```

Reply:
I got this from here: https://learn.microsoft.com/en-us/windows/uwp/gaming/directx-and-xaml-interop#surfaceimagesource

I tried sizeof and it crashed, so I will leave ARRAYSIZE.

Reply:
ARRAYSIZE is correct (number of elements).
sizeof is number of bytes, which is going to be # of elements * sizeof(each element), so you'll end up reading e.g. 7 * 4 entries instead of 7.

A better alternative to ARRAYSIZE (which is a macro, and non-standard) is std::size (still number of elements, doesn't rely on macros)

Reply:
Sorry, I mean '_countof'. Brain fart.